### PR TITLE
Add skills description

### PR DESCRIPTION
**Pull request by shravanhariharan2**
# Description

Add description field to skills for users to describe their skills.
Updated user queries to support updating and fetching of descriptions along w/ skill.

## Checklist

Remove any items which are not applicable.

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: app/database/models/usersModel.js [hunk 1]
```diff
@@ -94,7 +94,7 @@ async function findById(id) {
         'cons.point_of_contact_email',
         'cons.latitude',
         'cons.longitude',
-        db.raw('array_to_json(array_remove(array_agg(skills.skill), null)) as skills'),
+        db.raw('array_agg(json_build_object(\'skill\', skills.skill, \'description\', COALESCE(skills.description, \'\'))) as skills'),
```
Comment (dacruz21) [inline app/database/models/usersModel.js:97]: Does this remove nulls? I just fixed that bug yesterday. Test this on user id #5 in the seeded db.

Comment (shravanhariharan2) [reply app/database/models/usersModel.js:97]: Nope, it doesn't seem to remove nulls. I just found a solution to that bug so I'll push that up

**Comment by shravanhariharan2** [comment]
hmm I'm not too sure how to apply `array_remove` here, since I have to build the JSON first and I can't wrap the entire query in `array_remove` since I have to specify that `skills.skill`, which is part of a non-flat array, has to be null. Do you know if `array_remove` is possible to use here for arrays that aren't flat?

**Comment by dacruz21** [comment]
Ah I see, I didn't know that only `skills.skill` would be null. Let's just filter out the nulls within JS, it's a messy enough SQL statement as is.

**Comment by shravanhariharan2** [comment]
image

Yeah this is what the response looked like for user 5 without any filters, so rewriting in JS is probably a lot cleaner than adding to that super long SQL statement

**Pull request merged**